### PR TITLE
fix: log out from RH SSO if registry config or/and subs activation failed

### LINKED ISSUE #52: "LOGGED IN" despite of errors
While tracking down #42 on my Linux machine, I noticed that the extension claims to be logged in despite having failed to run subscription-manager.  The error is quite hidden in the tasks menu (which I expect most users to not know).

I think we need to make these error more obvious to users and not display to be LOGGED IN.  While technically we have the SSO token, some subsequent operations have failed, so users will interpret the green LOGGED IN as the operation to have been successful.

image

### PR DESCRIPTION
**Pull request by dgolovin**
The log out happens only after all commands failed or passed, so for example if registry access configuration failed and subscription activation still running log out will happen only when latter finished.

Subscription activation failed scenario

https://github.com/redhat-developer/podman-desktop-redhat-account-ext/assets/620330/fbcaaa61-eaf7-4002-9891-92975734ce8e

Registry configuration failed scenario

https://github.com/redhat-developer/podman-desktop-redhat-account-ext/assets/620330/a8fc2bec-f18c-4eef-8ba1-dc68a9ff766e

Fix #52.

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
LGTM, great work!

File: src/extension.ts [hunk 1]
```diff
@@ -292,7 +298,15 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
...
       }
-    );
+    ).then(() => false
+    ).catch(() => true); // required to force Promise.all() call keep waiting for all not failed calls
+  
+    const failed = await Promise.all([registryAccess,vmActivation]); // wait for all fail or pass
+    
+    if (failed.find(fail => fail)) {
+      removeSession(currentSession.id); // if at least one fail, remove session
+    }
```
Comment (benoitf) [inline src/extension.ts:308]: the code is OK but I think strange to have 'true' being the case where it failed and false being the case where it succeed.

Usually true = success, false = error

File: src/extension.ts [hunk 2]
```diff
@@ -292,7 +298,15 @@ export async function activate(extensionContext: extensionApi.ExtensionContext):
           }
         }
       }
-    );
+    ).then(() => false
+    ).catch(() => true); // required to force Promise.all() call keep waiting for all not failed calls
+  
+    const failed = await Promise.all([registryAccess,vmActivation]); // wait for all fail or pass
+    
+    if (failed.find(fail => fail)) {
```
Comment (benoitf) [inline src/extension.ts:306]: you could use `some()`

**Review by benoitf — APPROVED** [review]

File: src/extension.ts (hunk 1 above)
Comment (dgolovin) [reply src/extension.ts:308]: That is because I am looking for failure not for success.

File: src/extension.ts (hunk 2 above)
Comment (dgolovin) [inline src/extension.ts:306]:
```suggestion
    if (failed.some(fail => fail)) {
```

File: src/extension.ts (hunk 1 above)
Comment (benoitf) [reply src/extension.ts:308]: You could look for failure using false

Comment (benoitf) [reply src/extension.ts]: I mean array of success instead of array of failures

**Comment by vrothberg** [comment]
Thanks, @dgolovin !

**Pull request merged**